### PR TITLE
Add routes to get a single ingredient, and substitutions for a single…

### PR DESCRIPTION
**Pull request by elaguerta**
… ingredient

### CODE REVIEW

File: example/flaskr/ingredient.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+bp = Blueprint("ingredient", __name__, url_prefix="/ingredient")
+RATINGS = [ 'r_nourishment',
+            'r_value',
+            'r_human_welfare',
+            'r_animal_welfare',
+            'r_resource_cons',
+            'r_biodiversity',
+            'r_global_warming',]
+
+@bp.route("/<str:ingredientName>", methods=("GET", "POST"))
```
Comment (scheffeltravis) [inline example/flaskr/ingredient.py]: I would remove the POST method from this route if it's not being used.

File: example/flaskr/ingredient.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                "SELECT * FROM ingredient WHERE name = ?", ingredientName)
+                .fetchone()
+        
+        if ingredient is None:
+            abort(404, f"Ingredient with name {ingredientName} doesn't exist.")
+        
+        return ingredient
+
+
+@bp.route("/<str:ingredientName>/alt", methods=("GET"))
```
Comment (scheffeltravis) [inline example/flaskr/ingredient.py]: I think if a request is made to a route which does not allow any other routes (i.e. POST, DELETE, etc), it should get rejected with a 405 automatically: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

So I don't think you need to include the `methods` parameter here, as well as encompassing everything in the if statement (same with the other routes in this file).

**Review by scheffeltravis — APPROVED** [review]
🔥 killing it

**Comment by elaguerta** [comment]
This PR adds some basic routes to query the `ingredient` table for ingredient alternatives. 
1. `/ingredient/<ingredientName>/alt` displays all ingredients that are alternatives for `ingredientName` (including `ingredientName` itself). Below is `/ingredient/milk-generic/alt`:
image

2. `/ingredient/<ingredientName>/alt/<ratingName>` displays ingredients that are alternatives for `ingredientName` that have a higher rating for `ratingName`. Below is `ingredient/milk-generic/alt/r_biodiversity`, showing all equivalents to `milk-generic` that have a higher rating on `r_biodiversity`:
image

3. `/ingredient/<ingredientName>/alt/<ravg>` displays ingredients that alternatives for `ingredientName` that have a higher average rating over all ethical ratings than `ingredientName`'s average. Below is `/ingredient/milk-generic/ravg` showing equivalents to `milk-generic` with a higher average rating over all ethical ratings:
image

**Pull request merged**